sql/core/src/test/scala/org/apache/spark/sql/CollationSQLExpressionsSuite.scala

```diff
@@ -3260,6 +3260,39 @@ class CollationSQLExpressionsSuite
     }
   }
 
+  test("SPARK-50060: set operators with conflicting collations") {
```

Comment:
hmm, does this provide more test coverage than the golden files?

Reply:
Yes, I added also the test for collisions between default collation and explicit collation. In case we modify the behavior later, this test should also be changed.